launcher/verifier/rest/rest.go

```diff
@@ -0,0 +1,123 @@
+// Package rest contains the code to use the REST-based Google API
+package rest
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+
+	"github.com/google/go-tpm-tools/launcher/verifier"
+
+	v1alpha1 "google.golang.org/api/confidentialcomputing/v1alpha1"
+	"google.golang.org/api/option"
+)
+
+// Settings used to configure the REST client. Only one Client object should
+// be created per Project/Region pair.
+type Config struct {
+	Project string
+	Region  string
+	// Additional API options that can be used to override default settings.
+	Opts []option.ClientOption
+}
+
+func NewClient(ctx context.Context, cfg Config) (verifier.Client, error) {
+	service, err := v1alpha1.NewService(ctx, cfg.Opts...)
+	if err != nil {
+		return nil, fmt.Errorf("can't create ConfidentialComputing v1alpha1 API client: %v", err)
+	}
+	return &restClient{
+		parent:  fmt.Sprintf("projects/%v/locations/%v", cfg.Project, cfg.Region),
+		service: service}, nil
+}
+
+type restClient struct {
+	parent  string
+	service *v1alpha1.Service
+}
+
+var encoding = base64.StdEncoding
+
+// CreateChallenge implements verifier.Client
+func (c *restClient) CreateChallenge(ctx context.Context) (*verifier.Challenge, error) {
+	// Pass an empty Challenge for the input (all params are output-only)
+	output, err := c.service.Projects.Locations.Challenges.Create(
+		c.parent,
+		&v1alpha1.Challenge{},
+	).Do()
+	if err != nil {
+		return nil, fmt.Errorf("calling v1alpha1.CreateChallenge: %w", err)
+	}
+
+	nonce, err := encoding.DecodeString(output.Nonce)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode Challenge.Nonce: %w", err)
+	}
+
+	return &verifier.Challenge{
+		Name:  output.Name,
+		Nonce: nonce,
+	}, nil
+}
+
+// VerifyAttestation implements verifier.Client
+func (c *restClient) VerifyAttestation(ctx context.Context, request verifier.VerifyAttestationRequest) (*verifier.VerifyAttestationResponse, error) {
```

Comment:
given that the fields in `VerifyAttestationRequest` are exported, we should probably add a check that Challenge isn't nil

Reply:
and `Attestation` since `convertRequestToREST` assumes non-nil

Reply:
I changed the code to add a check, and also added tests that our conversion methods passed.